test/201_fullchain-multi-orders.test.ts

```diff
@@ -0,0 +1,255 @@
+let iexecWrapper: IexecWrapper;
+let [appAddress, workerpoolAddress, datasetAddress]: string[] = [];
+let [requester, appProvider, datasetProvider, scheduler, anyone, worker1]: SignerWithAddress[] = [];
+let ordersActors: OrdersActors;
+let ordersAssets: OrdersAssets;
+let ordersPrices: OrdersPrices;
+
+describe('Integration tests', function () {
+    beforeEach('Deploy', async () => {
+        // Deploy all contracts
+        proxyAddress = await loadHardhatFixtureDeployment();
+        // Initialize test environment
+        await loadFixture(initFixture);
+    });
+
+    async function initFixture() {
+        const accounts = await getIexecAccounts();
+        ({ requester, appProvider, datasetProvider, scheduler, anyone, worker1 } = accounts);
+        iexecWrapper = new IexecWrapper(proxyAddress, accounts);
+        ({ appAddress, datasetAddress, workerpoolAddress } = await iexecWrapper.createAssets());
+        iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, anyone);
+        iexecPocoAccessors = IexecPocoAccessors__factory.connect(proxyAddress, ethers.provider);
+        ordersActors = {
+            appOwner: appProvider,
+            datasetOwner: datasetProvider,
+            workerpoolOwner: scheduler,
+            requester: requester,
+        };
+        ordersAssets = {
+            app: appAddress,
+            dataset: datasetAddress,
+            workerpool: workerpoolAddress,
+        };
+        ordersPrices = {
+            app: appPrice,
+            dataset: datasetPrice,
+            workerpool: workerpoolPrice,
+        };
+    }
+
+    /**
+     * A test to run full workflow (matchOrders..finalize) with 2 orders having 2 different volumes
+     * for the same workerpool and only 1 request order.
+     */
+    it('[1] No sponsorship, no beneficiary, no callback, BoT, no replication, 2 workerpool orders', async function () {
+        const volume = 3;
+        const workerpoolOrderVolume1 = 2;
+        const workerpoolOrderVolume2 = 10;
+        const dealVolume1 = Math.min(workerpoolOrderVolume1, volume); // min(2, 3);
+        const dealVolume2 = Math.min(workerpoolOrderVolume2, volume - dealVolume1); // min(10, 1)
+        const workerpoolPrice1 = workerpoolPrice + 15;
+        const workerpoolPrice2 = workerpoolPrice + 25;
+        const taskPrice1 = appPrice + datasetPrice + workerpoolPrice1;
+        const taskPrice2 = appPrice + datasetPrice + workerpoolPrice2;
+        // Create default orders.
+        const {
+            appOrder,
+            datasetOrder,
+            workerpoolOrder,
+            requesterOrder: requestOrder,
+        } = buildOrders({
+            assets: ordersAssets,
+            prices: ordersPrices,
+            requester: requester.address,
+            tag: standardDealTag,
+            volume,
+        }).toObject();
+        // Create 2 different orders for the same workerpool.
+        const workerpoolOrder1 = workerpoolOrder;
```

Comment:
```suggestion
        const workerpoolOrder1 = { ...workerpoolOrder };
```
Maybe for mirroring calls